### PR TITLE
Update README, fix HLint CI

### PR DESCRIPTION
**Pull request by chshersh**
Adding some more information to README (our talk, known users).

Also attempted to fix the CI for HLint by pinning the HLint version and by downloading it from its binary releases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +58,12 @@ You will find Stan helpful if you enjoy writing in Haskell, but want
 more guarantees from your code, not provided by the Haskell type
 system or GHC.
 
+For a crash course to Stan, watch the talk about Stan, presented by
+Veronika Romashkina and Dmitrii Kovanikov at the Haskell Love
+conference.
+
+* [Stan – Haskell Static Analyser](https://www.youtube.com/watch?v=wb5PLv6-e6I)
```
Comment (vrom911) [inline README.md:65]: Do you think that embedding the vide would be a good solution?

```suggestion
[![Stan – Haskell Static Analyser](https://img.youtube.com/vi/wb5PLv6-e6I/0.jpg)](https://www.youtube.com/watch?v=wb5PLv6-e6I)
```

**Review by vrom911 — APPROVED** [review]
Great stuff!

File: README.md [hunk 2]
```diff
@@ -614,6 +621,13 @@ Our plan for the nearest future:
 We have much more ideas to work on.
 See more detailed plan in the dedicated [GitHub Project page](https://github.com/kowainik/stan/projects/1).
 
+## Users
+
+Stan is known to be used in the following companies:
```
Comment (vrom911) [inline README.md:626]: Alternative: adopted by, integrated to the companies' processes etc.?

Comment (chshersh) [reply README.md:626]: I like "Adopted by" 👍🏻 Does the following look okay?

```suggestion
Stan is known to be adopted by the following companies:
```

**Pull request merged**